intro2.md

```diff
@@ -0,0 +1,405 @@
+
+### What is Optimistic Rollup?
+
+[Optimistic rollup](https://vitalik.ca/general/2021/01/05/rollup.html) is a
+layer 2 scalability technique which increases the computation & storage capacity
+of Ethereum without sacrificing security or decentralization. Transaction data
+is submitted on-chain but executed off-chain. If there is an error in the
+off-chain execution, a fraud proof can be submitted on-chain to correct the
+error and protect user funds. In the same way you don't go to court unless there
+is a dispute, you don't execute transactions on on-chain unless there is an
+error. The rollup is *optimistic* because the execution results are assumed to
+be correct until a fraud proof proves them wrong.
+
+### What is EVM Equivalence?
+
+[EVM
+Equivalence](https://medium.com/ethereum-optimism/introducing-evm-equivalence-5c2021deb306)
+is complete compliance with the state transition function described in the
+Ethereum yellow paper, the formal definition of the protocol. By conforming to
+the Ethereum standard across EVM equivalent rollups, smart contract developers
+can write once and deploy anywhere.
+
+### 🎶 All together now 🎶
+
+#### Optimistic Ethereum is an _EVM equivalent_, _optimistic rollup_ protocol designed to _scale Ethereum_.
+
+## Network Participants
+
+There are three actors in Optimistic Ethereum: users, sequencers, and verifiers.
+
+<!--
+![Network Overview](./assets/network-participants-overview.svg)
+-->
+
+![Network Overview](https://raw.githubusercontent.com/ethereum-optimism/optimistic-specs/main/assets/network-participants-overview.svg)
+
+### Users
+
+At the heart of the network are users (us!). Users can:
+
+1. Deposit or withdraw tokens by sending transactions to Ethereum mainnet.
+2. Send transactions to the sequencer, to use EVM smart contracts on L2, or to
+   send ETH to other L2 users.
+3. View the status of transactions using block explorers provided by network
+   verifiers.
+
+### Sequencer
+
+As a first approximation, you can see the sequencer as the L2 block producer. It
+accrues user transactions into "batches" and submits these batches to contracts
+on L1 in order to make L2 blocks out of them (beware however, that there isn't a
+1-1 mapping between batches and blocks). Unlike on L1, on L2 transactions can be
+confirmed before a full block (or even a full batch) is created!
+
+We are going to give a lot more detail about the operation of the sequencer in
+the rest of this document.
+
+### Verifiers
+
+Verifiers monitor L1 for rollup data. They serve three purpose:
+
+1. Serving rollup data to users; and
+2. Verifying rollup integrity and disputing invalid assertions.
+3. Propagate the L2 state among validators (\*)
+
+In order for the network to remain secure there must be at least one honest
+verifier who is able to verify the integrity of the rollup chain & serve
+blockchain data to users.
+
+(\*) It's important to note that a validator does not rely on this data to
```

Comment:
I think this note is a bit too theoretical when it comes to the sequencer committing to the pre batched data. If I see a spec with how it could work in practice, then I'd be open to having it. I think for now its safe to say that the verifiers optimistically accept the data from the sequencer because users are interested in low latency transactions and that if the sequencer lies consistently then users will use a different, more reliable system.

We could hint at a cryptographically secure way of doing bond based sequencer pre-commitments if we wanted to though

Reply:
Re-reading this, it gives the impression that there are two classes of validators: those that just trust sequencer-signed L2 state propagation (but do not read L1), and those who just read L1 (to submit fraud proofs).

Would you say that's a correct assessment? (It seems suprising to me.)

If it's true, we should endeavour to remove that distinction, because one easy argument against the verifier's dilemma is that some actors *have to* run a verifier anyway (e.g. Etherscan), so we might as well have them truly *verify* the chain.